runtime/src/main/java/org/corfudb/runtime/collections/iSMRDeque.java

```diff
@@ -0,0 +1,139 @@
+package org.corfudb.runtime.collections;
+
+import org.corfudb.annotations.Accessor;
+import org.corfudb.annotations.Mutator;
+import org.corfudb.annotations.MutatorAccessor;
+
+import java.util.Deque;
+import java.util.Iterator;
+
+/**
+ * Double Ended Queue immutable supported by CorfuDB using State Machine Replication.
+ * Created by Sundar Sridharan on 5/8/19.
+ */
+public interface iSMRDeque<E> extends Deque<E>, ISMRObject {
```

Comment:
We do not want to implement Java.util interfaces. It becomes very onerous to maintain compatibility.

Reply:
Done. Removed.